hooks/open-telemetry/pkg/metrics.go

```diff
@@ -25,18 +25,10 @@ type MetricsHook struct {
 	errorCounter   api.Int64Counter
 
 	flagEvalMetadataDimensions []DimensionDescription
+	attributeMapperCallback    *func(openfeature.FlagMetadata) []attribute.KeyValue
```

Comment:
Hello 👋 I hope you don't mind me commenting on your PR, but I don't think you need to use a pointer here and it may be cleaner without it. The zero value of a function is nil.

Reply:
Thank you @craigpastro  for the review 

I updated the changes to accept nonpointer functions - https://github.com/open-feature/go-sdk-contrib/pull/311/commits/4c8b3d1e51b4cc3da09acc3895ec7310a8560b72

My initial reasoning for this was to have a comparability of the callback. But I think this is never going to be needed as we only have one callback registration. 

Also, thank you for mentioning documentation change need. I updated the readme accordingly